the-blue-alliance-ios/View Elements/Match/Breakdown/MatchBreakdownConfigurator2020.swift

```diff
@@ -0,0 +1,258 @@
+        guard let redEndgame = rw as? String, let blueEndgame = bw as? String else {
+            return nil
+        }
+
+        let elements = [redEndgame, blueEndgame].map { (endgame) -> [AnyHashable] in
+            let mode = UIView.ContentMode.center
+            if endgame == "IsLevel" {
+                let result: [AnyHashable] = [makeImageView(image: BreakdownStyle2020.checkImage, contentMode: mode), "(+15)"]
+                return result
+            }
+            let result: [AnyHashable] = [makeImageView(image: BreakdownStyle2020.xImage, contentMode: mode)]
+            return result
+        }
+        return BreakdownRow(title: title, red: elements.first ?? [], blue: elements.last ?? [])
+    }
+    
+    private static func shieldOperationalRow(title: String, red: [String: Any]?, blue: [String: Any]?) -> BreakdownRow? {
+        guard let shieldOperationalValues = values(key: "shieldOperationalRankingPoint", red: red, blue: blue) else {
+            return nil
+        }
+        let (rw, bw) = shieldOperationalValues
+        guard let redShieldOperational = rw as? Int, let blueShieldOperational = bw as? Int else {
+            return nil
+        }
+
+        let elements = [redShieldOperational, blueShieldOperational].map { (shieldOperational) -> [AnyHashable] in
+            if shieldOperational == 1 {
+                let result: [AnyHashable] = [makeImageView(image: BreakdownStyle2020.checkImage), "(+1 RP)"]
+                return result
+            }
+            let result: [AnyHashable] = [makeImageView(image: BreakdownStyle2020.xImage)]
+            return result
+        }
+        return BreakdownRow(title: title, red: elements.first ?? [], blue: elements.last ?? [])
+    }
+    
+    private static func stageActivationRow(title: String, red: [String: Any]?, blue: [String: Any]?) -> BreakdownRow? {
+        var redActivation: [Int] = [];
+        var blueActivation: [Int] = [];
+
+        for i in [3,2,1] {
+            guard let stageActivatedValues = values(key: "stage\(i)Activated", red: red, blue: blue) else {
+                return nil
+            }
+            let (rw, bw) = stageActivatedValues
+            guard let redStage = rw as? Int, let blueStage = bw as? Int else {
+                return nil
+            }
+            redActivation.append(redStage);
+            blueActivation.append(blueStage);
+        }
+
+        let elements = [redActivation, blueActivation].map { (stage) -> String in
+            if stage[0] == 1 {
+                return "3 (+1 RP)"
+            } else if stage[1] == 1 {
+                return "2"
+            } else if stage[2] == 1 {
+                return "1"
+            }
+            return ""
+        }
+        return BreakdownRow(title: title, red: [elements.first], blue: [elements.last])
+    }
+
+    private static func makeImageView(image: UIImage?, contentMode: UIView.ContentMode = UIView.ContentMode.scaleToFill) -> UIImageView {
```

Comment:
I love this. We should consider pulling this in to a place where it's shared. Maybe make it take a variable tint color too. We have something similar for labels in `BreakdownStyle` (Don't worry about refactoring old UIImageView creation - I can make an issue to do that)

Reply:
Thanks!  In 7433525, I moved it to `BreakdownStyle.imageView`, with an added optional tint, which defaults to `UIColor.label`, for easier use,